### PR TITLE
#6840 Report useless array filter

### PR DESCRIPTION
**Pull request by leongersen**
(I don't have access to PHP8 right now, I'll have to fix the build in the CLI)

Implements phpstan/phpstan#6840

### CODE REVIEW

File: tests/PHPStan/Rules/Functions/data/array_filter_empty.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+/** @var \stdClass[] $objects */
```
Comment (staabm) [inline tests/PHPStan/Rules/Functions/data/array_filter_empty.php:3]: please add a namespace to this file and move the testing code into a method of a class with a name of your choice.

File: src/Rules/Functions/ArrayFilterEmptyRule.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	{
+		if (!($node->name instanceof Node\Name)) {
+			return [];
+		}
+
+		if ($this->reflectionProvider->resolveFunctionName($node->name, $scope) !== 'array_filter') {
+			return [];
+		}
+
+		$args = array_values($node->getArgs());
```
Comment (staabm) [inline src/Rules/Functions/ArrayFilterEmptyRule.php:42]: is the `array_values` call necessary?

File: src/Rules/Functions/ArrayFilterEmptyRule.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		return FuncCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!($node->name instanceof Node\Name)) {
+			return [];
+		}
+
+		if ($this->reflectionProvider->resolveFunctionName($node->name, $scope) !== 'array_filter') {
```
Comment (staabm) [inline src/Rules/Functions/ArrayFilterEmptyRule.php:38]: this should be case-insensitve, e.g. using `strtolower`

**Comment by staabm** [comment]
I guess this PR should target the 1.4.x branch

**Comment by leongersen** [comment]
I've created a new PR to 1.4.x: #1077

**Comment by herndlm** [comment]
Not entirely sure, if it really should target the 1.4.x branch, but regarding the rebase: I think an interactive rebase would have worked where you have to manually drop all the new commits from 1.5.x. In case you need to do something like that again :)

**Comment by ondrejmirtes** [comment]
Hi, it's unfortunate but you should always evaluate every review with critical thinking :) This is absolutely a new feature, hence it should target 1.5.x

**Comment by leongersen** [comment]
No problem! I'm learning things along the way 😄 . I can't reopen this PR, but I've rebased the new one to 1.5.x.

image

**Comment by staabm** [comment]
> Hi, it's unfortunate but you should always evaluate every review with critical thinking :) This is absolutely a new feature, hence it should target 1.5.x


Ok it seems we have a different understanding about the branches and what is considered a feature or not.

IMO https://github.com/phpstan/phpstan-src/pull/1067  was also a new feature 🤷‍♂️

**Pull request closed without merging**